File: src/com/ikraybill/chess/gui/DebugHud.java
package com.ikraybill.chess.gui;

import java.awt.*;
import java.util.HashMap;
import java.util.Map;

public class DebugHud {
    private String renderedText;
    private Map<String, Object> content;
    private int fontSize;

    public DebugHud(){
        content = new HashMap<>();
    }

    public void addLine(String description, Object data){
        content.put(description, data);
        renderedText = "";
        for (Map.Entry<String, Object> me: content.entrySet()) {
            renderedText += me.getKey() + ": " + me.getValue().toString() + "\n";
        }

    }

    public void removeLine(String description){
        content.remove(description);
        renderedText = "";
        for (Map.Entry<String, Object> me: content.entrySet()) {
            renderedText += me.getKey() + ": " + me.getValue().toString() + "\n";
        }
    }

    //TODO
    public void draw(Graphics g){
        int i = 1;
        System.out.println(i);
        g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, fontSize));
        for (Map.Entry<String, Object> me: content.entrySet()) {
            i++;
            g.drawString(me.getKey() + ": " + me.getValue().toString(), 10, fontSize * i);
        }
    }
}

File: src/com/ikraybill/chess/gui/DebugHud.kt
package com.ikraybill.chess.gui

import java.awt.*
import java.util.HashMap
import javax.swing.JPanel

class DebugHud: JPanel(true) {
    private var renderedText: String? = null
    private val content: MutableMap<String, Any>

    init {
        content = HashMap()
    }

    fun addLine(description: String, data: Any) {
        content[description] = data
        renderedText = ""
        for ((key, value) in content) {
            renderedText += key + ": " + value.toString() + "\n"
        }

    }

    fun removeLine(description: String) {
        content.remove(description)
        renderedText = ""
        for ((key, value) in content) {
            renderedText += key + ": " + value.toString() + "\n"
        }
    }

    fun draw(g: Graphics) {
        var i = 1
        val fontSize = 16
        g.font = Font(Font.SANS_SERIF, Font.BOLD, fontSize)
        for ((key, value) in content) {
            i++
            g.drawString(key + ": " + value.toString(), 10, fontSize * i)
        }
    }

    @Synchronized
    override fun paint(g: Graphics){
        super.paint(g)
        var i = 1
        val fontSize = 16
        g.color = Color(70, 73, 255)
        g.font = Font(Font.SANS_SERIF, Font.BOLD, fontSize)
        for ((key, value) in content) {
            i++
            g.drawString(key + ": " + value.toString(), 10, fontSize * i)
        }
    }
}
